fiftyone/brain/internal/core/leaky_splits.py

```diff
@@ -0,0 +1,435 @@
+
+# pylint: disable=no-member
+import cv2
+
+import eta.core.utils as etau
+
+import fiftyone as fo
+import fiftyone.core.brain as fob
+import fiftyone.brain.similarity as sim
+import fiftyone.brain.internal.core.sklearn as skl_sim
+import fiftyone.brain.internal.core.duplicates as dups
+import fiftyone.core.utils as fou
+import fiftyone.core.validation as fov
+import fiftyone.zoo as foz
+
+fbu = fou.lazy_import("fiftyone.brain.internal.core.utils")
+
+_DEFAULT_MODEL = "clip-vit-base32-torch"
+_DEFAULT_BATCH_SIZE = None
+
+
+def compute_leaky_splits(
+    samples,
+    brain_key=None,
+    split_views=None,
+    split_field=None,
+    split_tags=None,
+    threshold=0.2,
+    similarity_brain_key=None,
+    embeddings_field=None,
+    model=None,
+    model_kwargs=None,
+    similarity_backend=None,
+    similarity_config_dict=None,
+    **kwargs,
+):
+    """See ``fiftyone/brain/__init__.py``."""
+
+    fov.validate_collection(samples)
+
+    config = LeakySplitsConfig(
+        split_views,
+        split_field,
+        split_tags,
+        similarity_brain_key,
+        embeddings_field,
+        model,
+        model_kwargs,
+        similarity_backend,
+        similarity_config_dict,
+        **kwargs,
+    )
+    brain_method = config.build()
+    brain_method.ensure_requirements()
+
+    if brain_key is not None:
+        # Don't allow overwriting an existing run with same key, since we
+        # need the existing run in order to perform workflows like
+        # automatically cleaning up the backend's index
+        brain_method.register_run(samples, brain_key, overwrite=False)
+
+    results = brain_method.initialize(samples, brain_key)
+    results.set_threshold(threshold)
+    leaks = results.leaks
+
+    if brain_key is not None:
+        brain_method.save_run_results(samples, brain_key, results)
+    if results._save_similarity_index:
+        results._similarity_method.save_run_results(
+            samples, similarity_brain_key, results._similarity_index
+        )
+
+    return results, leaks
+
+
+### GENERAL
+
+
+class LeakySplitsConfig(fob.BrainMethodConfig):
+    def __init__(
+        self,
+        split_views=None,
+        split_field=None,
+        split_tags=None,
+        similarity_brain_key=None,
+        embeddings_field=None,
+        model=None,
+        model_kwargs=None,
+        similarity_backend=None,
+        similarity_config_dict=None,
+        **kwargs,
+    ):
+        self.split_views = split_views
+        self.split_field = split_field
+        self.split_tags = split_tags
+        self.similarity_brain_key = similarity_brain_key
+        self.embeddings_field = embeddings_field
+        self.model = model
+        self.model_kwargs = model_kwargs
+        self.similarity_backend = similarity_backend
+        self.similarity_config_dict = similarity_config_dict
+        super().__init__(**kwargs)
+
+    @property
+    def type(self):
+        return "Data Leakage"
```

Comment:
For consistency with existing code, let's keep it lowercase one word: `leakage`?

Reply:
changed